build.rs

```diff
@@ -0,0 +1,116 @@
+use std::io::{Write, BufWriter};
+
+// Location of the dist folder
+#[cfg(not(target_os = "windows"))]
+const DIST: &str = "frontend/dist/";
+#[cfg(target_os = "windows")]
+const DIST: &str = "frontend\\dist\\";
+
+// Where to write the Assets out to
+#[cfg(not(target_os = "windows"))]
+const ASSETOUT: &str = "src/asset.in";
+#[cfg(target_os = "windows")]
+const ASSETOUT: &str = "src\\asset.in";
+
+// Ignore directories specified here
+#[cfg(target_os = "windows")]
+lazy_static! {
+    static ref IGNORE: Vec<&'static str> = {
+        vec![
+            "frontend\\dist\\ember-fetch"
+        ]
+    };
+}
+#[cfg(not(target_os = "windows"))]
+lazy_static! {
+    static ref IGNORE: Vec<&'static str> = {
+        vec![
+            "frontend/dist/ember-fetch"
+        ]
+    };
+}
+
+/// Find the assets and write out the corresponding files
+pub fn main() {
+    write_asset_file(acquire_assets());
+}
+
+/// Start the recursion from the top level of the frontend's dist folder
+fn acquire_assets() -> Vec<Asset> {
+    recurse_dir(&Path::new(DIST))
+}
+
+/// Recurse through a directory, if it's an asset file add it to our list and if
+/// it's a directory get all the assets in there then put the `Assets` returned
+/// into this directory's Vector and return that.
+fn recurse_dir<P>(path: &P) -> Vec<Asset>
+where
+    P: AsRef<Path>,
+{
+    let mut output: Vec<Asset> = Vec::new();
+
+    for entry in read_dir(path.as_ref()).expect("Unable to read dist folder") {
+        let entry = entry.expect("Unable to read directory entry");
+        let meta = metadata(&entry.path()).expect("Unable to get metadata on entry");
+
+        if !IGNORE.contains(&entry.path().to_str().unwrap()) {
+            if meta.is_dir() {
+                output.extend(recurse_dir(&entry.path()));
+            } else {
+                output.push(Asset {
+                    // If the directory isn't valid this wouldn't have worked.
+                    path: String::from(entry.path().to_str().unwrap()),
+                });
+            }
+        }
+    }
+
+    output
+}
+
+/// Write to asset.in which is a vec expression of static assets that will be
+/// imported using the include!() macro for the `Config` struct
+fn write_asset_file(assets: Vec<Asset>) {
+    let mut writer = BufWriter::new(File::create(ASSETOUT).expect("Unable to create file"));
+    writer.write_all(b"vec![").unwrap();
+    for i in assets {
+        writer.write_fmt(format_args!("{},", i)).unwrap();
+    }
+    writer.write_all(b"]\n").unwrap();
+}
+
+/// Intermediary data structure used to hold the path of the file
+pub struct Asset {
+    pub path: String,
+}
+
+// This gives us the ToString impl as well. We're using this to write out what
+// an asset would look like for the actual program into the asset.in file.
+impl fmt::Display for Asset {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut buffer = String::from("Asset { name: \"");
```

Comment:
Have a look at https://doc.rust-lang.org/nightly/std/fmt/struct.Formatter.html#method.debug_struct and its friends :)